src/engine/effects/engineeffectchain.cpp

```diff
@@ -165,98 +165,77 @@ void EngineEffectChain::process(const ChannelHandle& handle,
 
     EffectProcessor::EnableState effectiveEnableState = channel_info.enable_state;
 
+    if (channel_info.enable_state == EffectProcessor::DISABLING) {
+        channel_info.enable_state = EffectProcessor::DISABLED;
```

Comment:
I outlined a general idea at the end of the comments on that bug, but there are details that remain unresolved. Anyway, this PR does not introduce the issue, so there's no need this PR needs to solve it.